### ISSUE TITLE
DynamoDB: `batch_get_item()` no longer works with `Binary` values

### ISSUE THREAD

**Issue by jrobbins-LiveData** [getmoto/moto#6827]
`moto` version `4.2.3` (due to #6816) `batch_get_item()` no longer works with `Binary` values.

We edited [test_dynamodb_batch_get_item.py](https://github.com/getmoto/moto/blob/master/tests/test_dynamodb/test_dynamodb_batch_get_item.py) to create the test table with `Binary` values and the modified test copied here reproduces the crash:

```python
import base64

import boto3
from moto import mock_dynamodb


def _create_user_table():
    client = boto3.client("dynamodb", region_name="us-east-1")
    client.create_table(
        TableName="users",
        KeySchema=[{"AttributeName": "username", "KeyType": "HASH"}],
        AttributeDefinitions=[{"AttributeName": "username", "AttributeType": "S"}],
        ProvisionedThroughput={"ReadCapacityUnits": 5, "WriteCapacityUnits": 5},
    )
    client.put_item(
        TableName="users", Item={"username": {"S": "user1"}, "foo": {"B": base64.b64encode(b"bar")}}
    )
    client.put_item(
        TableName="users", Item={"username": {"S": "user2"}, "foo": {"B": base64.b64encode(b"bar")}}
    )
    client.put_item(
        TableName="users", Item={"username": {"S": "user3"}, "foo": {"B": base64.b64encode(b"bar")}}
    )
    return client


@mock_dynamodb
def test_batch_items_returns_all():
    dynamodb = _create_user_table()
    returned_items = dynamodb.batch_get_item(
        RequestItems={
            "users": {
                "Keys": [
                    {"username": {"S": "user0"}},
                    {"username": {"S": "user1"}},
                    {"username": {"S": "user2"}},
                    {"username": {"S": "user3"}},
                ],
                "ConsistentRead": True,
            }
        }
    )["Responses"]["users"]
    assert len(returned_items) == 3
    assert [item["username"]["S"] for item in returned_items] == [
        "user1",
        "user2",
        "user3",
    ]
```

Partial traceback (eliding the many lines of `boto3` nested calls)
```
.venv\lib\site-packages\moto\dynamodb\responses.py:635: in batch_get_item
    return dynamo_json_dump(results)
.venv\lib\site-packages\moto\dynamodb\models\utilities.py:13: in dynamo_json_dump
    return json.dumps(dynamo_object, cls=DynamoJsonEncoder)
C:\Users\jeff1\.pyenv\pyenv-win\versions\3.10.11\lib\json\__init__.py:238: in dumps
    **kw).encode(obj)
C:\Users\jeff1\.pyenv\pyenv-win\versions\3.10.11\lib\json\encoder.py:199: in encode
    chunks = self.iterencode(o, _one_shot=True)
C:\Users\jeff1\.pyenv\pyenv-win\versions\3.10.11\lib\json\encoder.py:257: in iterencode
    return _iterencode(o, 0)
.venv\lib\site-packages\moto\dynamodb\models\utilities.py:9: in default
    return o.to_json()
.venv\lib\site-packages\moto\dynamodb\models\dynamo_type.py:202: in to_json
    return {self.type: base64.b64encode(self.value).decode("utf-8")}
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _

s = 'WW1GeQ==', altchars = None

    def b64encode(s, altchars=None):
        """Encode the bytes-like object s using Base64 and return a bytes object.

        Optional altchars should be a byte string of length 2 which specifies an
        alternative alphabet for the '+' and '/' characters.  This allows an
        application to e.g. generate url or filesystem safe Base64 strings.
        """
>       encoded = binascii.b2a_base64(s, newline=False)
E       TypeError: a bytes-like object is required, not 'str'
```

**Comment by dariocurr** [comment]
Same for me. I guess the fix for #6816 broke something else

**Comment by bblommers** [comment]
Thanks for raising this @jrobbins-LiveData - looks like I didn't do enough testing there. Will submit a fix shortly.

**Comment by bblommers** [comment]
The same problem also manifested itself when calling `scan` or `query` - that should all be fixed in moto >= 4.2.4.dev6.

Please let us know if you run into any other issues.
